### ISSUE TITLE
CA1033 is reported for interfaces suggesting to seal them

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslyn-analyzers#6785]
### Analyzer

**Diagnostic ID**: CA1033: Interface methods should be callable by child types

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview1.23165.1 (Latest)

### Steps To Reproduce

```csharp
public interface IConsumer
{
    Task ConsumeAsync();
}

public interface ISyncConsumer : IConsumer
{
    Task IConsumer.ConsumeAsync() // <- suggest to make ISyncConsumer sealed
    {
        Consume();

        return Task.CompletedTask;
    }

    void Consume();
}
```

### Expected behavior

No warning is reported

### Actual behavior

CA1033 with the weird message is reported

**Comment by sharwell** [comment]
The issue described by the documentation for CA1033 appears to apply even for the situation described above, so this does not appear to be a false positive.

**Comment by hankovich** [comment]
OK, but at least it should not suggest me to seal the interface.

**Comment by sharwell** [comment]
Yes, we'll have to figure out what the expected wording here is.

**Comment by Alex-Sob** [comment]
@sharwell I believe that this rule does not apply to interfaces. The issue arises when there's a class `A` that implements interface `I` explicitly, and there's class `B` that derives from `A` and also implements `I`. And the issue is that `B` cannot call base implementation.
In case of interfaces, the method is the default implementation that is used only when implementing type does not implement that method itself. So it's not supposed to be called explicitly. Probably the analyzer had existed before default interface implementations were introduced and was not updated for the new feature.
